cmd/agent/main.go

```diff
@@ -1,3 +1,124 @@
 package main
 
-func main() {}
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"runtime"
+	"strings"
+	"time"
+)
+
+type (
+	gauge      float64
+	counter    uint64
+	metricsMap map[string]any
+)
+
+const (
+	pollInterval   = 2 * time.Second
+	reportInterval = 10 * time.Second
+)
+
+var (
+	g             gauge
+	c             counter
+	m             runtime.MemStats
+	listOfMetrics metricsMap
+	tickerpoll    = time.NewTicker(pollInterval)
+	tickerreport  = time.NewTicker(reportInterval)
+)
+
+func collectMetrics(metricList metricsMap) metricsMap {
```

Comment:
с мапой можно использовать грязный хак.
Так как под капотом данные хранятся по ссылке, то не обязательно возвращать саму мапу, данные будут обновлены в исходной мапе и тогда уйдут все танцы с двумя объявлениями мапы.
```suggestion
func collectMetrics(metricList metricsMap) {
```